csrc/scheduler/registry.cpp

```diff
@@ -1981,7 +1975,7 @@ class PersistentKernelScheduler : public SchedulerEntry {
       return false;
     }
 
-    const int64_t max_used_sms = is_cross_grid
+    const int64_t max_used_sms = is_cross_grid && cross_grid_params.has_value()
```

Comment:
Why is this changed?

Reply:
There's an assertion inside a if block with `is_cross_grid`, so it's guaranteed to have a value when `is_cross_grid`. If that's not enough for clang-tidy, let's put an assertion again since it doesn't make any sense to have `is_cross_grid` being `true` and `cross_grid_params.has_value()` being `false`.